binary_search_tree/binary_search_tree.py

```diff
@@ -1,61 +1,136 @@
+                self.left = BSTNode(value)
+            else:
+                # We hop over the node that the instance is pointing to and insert new node recursively
+                self.left.insert(value)
+        else:
+            # If the value of the instance was greater
+            # We then check if the  instances right pointer is equal to None
+            if self.right is None:
+                # If it is we then instantiate a new instance and set the initial instances right pointer to the new instance
+                self.right = BSTNode(value)
+            else:
+                # We hop over the node that the instance is pointing to and insert new node recursively
+                self.right.insert(value)
 
     # Return True if the tree contains the value
     # False if it does not
+
     def contains(self, target):
-        pass
+        # If initial instance value is the same as the new value return true
+        if self.value is target:
+            return True
+        # We then decide which direction to traverse down the tree
+        # If this is true we go left and check the values
+        if self.value > target:
+            if self.left is not None:
+                return self.left.contains(target)
+        # If this is true we go right and check the values
+        elif self.value < target:
+            if self.right is not None:
+                return self.right.contains(target)
+        else:
+            # If neither of the previous conditions are met we return False
+            return False
 
     # Return the maximum value found in the tree
+
     def get_max(self):
-        pass
+        if self.right == None:
+            return self.value
+        else:
+            return self.right.get_max()
 
     # Call the function `fn` on the value of each node
     def for_each(self, fn):
-        pass
+        base = self.value
+        if not self.left and not self.right:
+            return fn(base)
+        if self.left and not self.right:
+            return fn(base), self.left.for_each(fn)
+        if self.right and not self.left:
+            return fn(base), self.right.for_each(fn)
+        if self.right and self.left:
+            return fn(base), self.right.for_each(fn), self.left.for_each(fn)
 
     # Part 2 -----------------------
 
     # Print all the values in order from low to high
     # Hint:  Use a recursive, depth first traversal
     def in_order_print(self, node):
-        pass
+        if node:
+            self.in_order_print(node.left)
+            print(node.value)
+            self.in_order_print(node.right)
+
+ # Print the value of every node, starting with the given node,
+ # in an iterative breadth first traversal
 
-    # Print the value of every node, starting with the given node,
-    # in an iterative breadth first traversal
     def bft_print(self, node):
-        pass
+        queue = []
+        queue.append(node)
+
+        while len(queue) > 0:
+            node = queue.pop(0)
```

Comment:
Great work! Remember that popping at index 0 is O(N) for lists. I think it would be best to use collections.deque the same way we talked about in stand up today. It is a doubly-linked list underneath the hood so removing anything from the front of the list is O(1).